File: Javascript/26.remove-duplicates-from-sorted-array.js
/*
 * @lc app=leetcode id=26 lang=javascript
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    const set = new Set(nums);
    nums.splice(0, nums.length);
    set.forEach(v => {
        nums.push(v);
    });

    return set.size;
};
// @lc code=end

File: Ruby/26.remove-duplicates-from-sorted-array.rb
#
# @lc app=leetcode id=26 lang=ruby
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    map = {}
    length = nums.size

    0.upto(length - 1) do |i|
        if map[nums[i]]
            nums[i] = nil
        else
            map[nums[i]] = nums[i]
        end
    end

    nums.compact!
    nums.size
end
# @lc code=end

